thoth-api/src/model/file/crud.rs

```diff
@@ -0,0 +1,269 @@
+        self.file_id
+    }
+
+    fn all(
+        _db: &crate::db::PgPool,
+        _limit: i32,
+        _offset: i32,
+        _filter: Option<String>,
+        _order: Self::OrderByEntity,
+        _publishers: Vec<Uuid>,
+        _parent_id_1: Option<Uuid>,
+        _parent_id_2: Option<Uuid>,
+        _filter_param_1: Vec<Self::FilterParameter1>,
+        _filter_param_2: Vec<Self::FilterParameter2>,
+        _filter_param_3: Option<Self::FilterParameter3>,
+        _filter_param_4: Option<Self::FilterParameter4>,
+    ) -> ThothResult<Vec<File>> {
+        Err(ThothError::InternalError(
+            "File::all not implemented".to_string(),
+        ))
+    }
+
+    fn count(
+        _db: &crate::db::PgPool,
+        _filter: Option<String>,
+        _publishers: Vec<Uuid>,
+        _filter_param_1: Vec<Self::FilterParameter1>,
+        _filter_param_2: Vec<Self::FilterParameter2>,
+        _filter_param_3: Option<Self::FilterParameter3>,
+        _filter_param_4: Option<Self::FilterParameter4>,
+    ) -> ThothResult<i32> {
+        Err(ThothError::InternalError(
+            "File::count not implemented".to_string(),
+        ))
+    }
+
+    fn from_id(db: &crate::db::PgPool, entity_id: &Uuid) -> ThothResult<Self> {
+        use diesel::QueryDsl;
+        use diesel::RunQueryDsl;
+        let mut connection = db.get()?;
+        file::table
+            .find(entity_id)
+            .get_result::<File>(&mut connection)
+            .map_err(|e: diesel::result::Error| ThothError::from(e))
+    }
+
+    fn create(db: &crate::db::PgPool, data: &NewFile) -> ThothResult<Self> {
+        let mut connection = db.get()?;
+        diesel::insert_into(file::table)
+            .values(data)
+            .get_result::<File>(&mut connection)
+            .map_err(|e: diesel::result::Error| ThothError::from(e))
+    }
+
+    fn update(
+        &self,
+        _db: &crate::db::PgPool,
+        _data: &NewFile,
+        _account_id: &Uuid,
+    ) -> ThothResult<Self> {
+        Err(ThothError::InternalError(
+            "File::update not implemented".to_string(),
+        ))
+    }
+
+    fn delete(self, db: &crate::db::PgPool) -> ThothResult<Self> {
+        use diesel::QueryDsl;
+        use diesel::RunQueryDsl;
+        let mut connection = db.get()?;
+        diesel::delete(file::table.find(self.file_id))
+            .execute(&mut connection)
+            .map(|_| self)
+            .map_err(|e: diesel::result::Error| ThothError::from(e))
+    }
+
+    fn publisher_id(&self, db: &crate::db::PgPool) -> ThothResult<Uuid> {
+        match (self.work_id, self.publication_id) {
+            (Some(work_id), None) => {
+                crate::model::work::Work::from_id(db, &work_id)?.publisher_id(db)
+            }
+            (None, Some(publication_id)) => {
+                crate::model::publication::Publication::from_id(db, &publication_id)?
+                    .publisher_id(db)
+            }
+            _ => Err(ThothError::InternalError(
```

Comment:
I've seen a bunch of `ThothError::InternalError` in the PR, but these are not internal errors but user input errors... So let's just create new variants for each of them in `thoth-errors`